libs/azure-api-client/src/Authentication/AuthenticatorFactory.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\AzureApiClient\Authentication;
+
+use Keboola\AzureApiClient\Exception\ClientException;
+use Keboola\AzureApiClient\GuzzleClientFactory;
+
+class AuthenticatorFactory
+{
+    public function getAuthenticator(GuzzleClientFactory $clientFactory): AuthenticatorInterface
+    {
+        $authenticator = new ClientCredentialsEnvironmentAuthenticator($clientFactory);
+        try {
+            $authenticator->checkUsability();
+            return $authenticator;
+        } catch (ClientException $e) {
+            $clientFactory->getLogger()->debug(
+                'ClientCredentialsEnvironmentAuthenticator is not usable: ' . $e->getMessage()
+            );
+        }
+        /* ManagedCredentialsAuthenticator checkUsability method has poor performance due to slow responses
```

Comment:
tohle je přesně ten důvod proč si nemyslím, že by checkUsability mělo být součástí interfacu, tenhle záměr se neosvědčíl

Reply:
Tohle je 1:1 prevzaty z https://github.com/keboola/azure-key-vault-php-client/blob/master/src/Authentication/AuthenticatorFactory.php. Netvrdim ze s emi to libi, ale improvements bych nechal na jindy.

Reply:
ok